src/backend/api/meals.js

```diff
@@ -1,15 +1,133 @@
-const express = require("express");
+const express = require('express');
+const { andHaving } = require('../database');
 const router = express.Router();
-const knex = require("../database");
-
-router.get("/", async (request, response) => {
-  try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
-  } catch (error) {
-    throw error;
-  }
+const knex = require('../database');
+
+router.post('/', async (request, response) => {
+    try {
+        // knex syntax for selecting things.
+        const insertMeal = await knex('meal').insert({
+            title: request.body.title,
+            description: request.body.description,
+            location: request.body.location,
+            when: request.body.when,
+            max_reservations: request.body.max_reservations,
+            price: request.body.price,
+            created_date: request.body.created_date,
+        });
+        response.json(insertMeal);
+    } catch (error) {
+        throw error;
+    }
+});
+
+router.put('/:id', async (request, response) => {
+    try {
+        const titles = await knex('meal')
+            .where({ id: request.params.id })
+            .update(request.body);
+        response.json(titles);
+    } catch (error) {
+        throw error;
+    }
+});
+
+router.delete('/:id', async (request, response) => {
+    try {
+        const titles = await knex('meal')
+            .where({ id: request.params.id })
+            .delete(request.body);
+        response.json(titles);
+    } catch (error) {
+        throw error;
+    }
+});
+
+router.get('/:id', async (request, response) => {
+    try {
+        const titles = await knex('meal').where({ id: request.params.id });
+        response.json(titles);
+    } catch (error) {
+        throw error;
+    }
+});
+
+//api/meals?maxPrice=90
+router.get('/', async (request, response) => {
```

Comment:
While it works as expected there a few things that could be done better and if you're up for it you can improve it:
- The route will not work with multiple parameters at the same time as expected. If for instance I pass in both 'maxPrice' and 'availableReservations' you will return meals filtered only by the MaxPrice parameters since that is the one you check first.
- You don;t have to use twice the 'await' keyword
- Check one more comment below for the isNaN check

I strongly suggest you watch the following [video ](https://www.youtube.com/watch?v=4KV79W4g2QU&t=418s)made by Orhan at HYF which explains how you can achieve the optimal behavior with the multiple params.

Reply:
I fixed it and the route is working with multiple parameters now. Is the solution ok?

Reply:
Looks great, nice job :100: